src/paperqa/docs.py

```diff
@@ -781,26 +781,53 @@ async def aquery(  # noqa: PLR0912
             # Only keep "\nFrom {citation}" if we are showing detailed citations
             context_inner_prompt = context_inner_prompt.replace("\nFrom {citation}", "")
 
-        inner_context_strs = [
-            context_inner_prompt.format(
-                name=c.id,
-                text=c.context,
-                citation=c.text.doc.formatted_citation,
-                **(c.model_extra or {}),
-            )
-            for c in filtered_contexts
-        ]
+        context_str_body = ""
+        if answer_config.group_contexts_by_question:
+            contexts_by_question: dict[str, list] = {}
```

Comment:
Can you make this a `defaultdict(list)` to remove the need for this below:

```python
if question not in contexts_by_question:
    contexts_by_question[question] = []
```

Also, can you populate `list[Context]` in the type hint